src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -137,6 +137,16 @@ is_SC_fixed(unsigned int index)
   }
 }
 
+// Function to print progress using dots
+static void
+progressReporter(bool& done)
```

Comment:
Are you aware of the core/tools/common/ProgressBar.cpp functionality.   

This class will perform this operation for you.

Reply:
Hi @rozumx, yes I have checked ProgressBar.cpp file initially and tried integrating this code but there we need to feed status of the task every time using update function but at present we don't get any progress status from VMR it just returns success or failure. So created a new static function to print dots until success or failure status is returned.

If we need to use ProgressBar.cpp code one way I can think of is to set max iterations as value greater than time out from VMR and call update function every time after some fixed duration of time.